plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/JsonFileGModelStore.java

```diff
@@ -39,36 +44,31 @@
 import com.google.inject.Inject;
 
 /**
- * A source model loader that reads the graph model directly from a JSON file.
+ * A source model storage that reads and writes the graph model directly from and to a JSON file.
  */
-public class JsonFileGModelLoader implements ModelSourceLoader {
+public class JsonFileGModelStore implements SourceModelStorage {
 
-   private static Logger LOG = Logger.getLogger(JsonFileGModelLoader.class);
+   private static Logger LOG = Logger.getLogger(JsonFileGModelStore.class);
    private static String EMPTY_ROOT_ID = "glsp-graph";
 
    @Inject
-   private GraphGsonConfigurationFactory gsonConfiguratior;
+   private GraphGsonConfigurationFactory gsonConfigurator;
 
    @Inject
    protected GModelState modelState;
 
    @Override
    public void loadSourceModel(final RequestModelAction action) {
-      final File file = convertToFile(modelState);
+      final File file = convertToFile(action.getOptions());
       loadSourceModel(file, modelState).ifPresent(root -> {
          modelState.setRoot(root);
          modelState.getRoot().setRevision(-1);
       });
    }
 
-   protected File convertToFile(final GModelState modelState) {
-      return getOrThrow(ClientOptionsUtil.getSourceUriAsFile(modelState.getClientOptions()),
-         "Invalid file URI:" + ClientOptionsUtil.getSourceUri(modelState.getClientOptions()));
-   }
-
    protected Optional<GModelRoot> loadSourceModel(final File file, final GModelState modelState) {
       try (Reader reader = new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8)) {
-         Gson gson = gsonConfiguratior.configureGson().create();
+         Gson gson = gsonConfigurator.configureGson().create();
```

Comment:
I'm wondering whether it's really necessary to construct a new Gson instance for every load/save request.
With method injection we could create a shared instance during injection time: 
```java
protected Gson gson;
@Inject
public void configureGson(GraphGsonConfiguratorFactory gsonConfigurator) {
gson =  gsonConfigurator.configureGson().setPrettyPrinting().create();
}
```